src/Og.php

```diff
@@ -196,17 +194,17 @@ public static function getEntityGroups(EntityInterface $entity, array $states =
 
     /** @var \Drupal\og\Entity\OgMembership $membership */
     foreach ($memberships as $membership) {
-      static::$entityGroupCache[$identifier][$membership->getGroupEntityType()][$membership->id()] = $membership->getGroup();
+      static::$cache[$identifier][$membership->getGroupEntityType()][$membership->id()] = $membership->getGroup();
```

Comment:
> Then again, if you query the memberships, it's probably because you are interested more in the groups than in the memberships themselves.

In D7 when we call `og_get_entity_groups()` we sometimes need the memberships, and sometimes the groups. I agree it could be cleaner to split it into two different methods.

regarding the static cache - since we know that `Entity::loadMultiple` already caches the object, maybe it would make more sense to only keep the IDs, so we don't increase too much the memory consumption?